### PR TITLE
JAMES-2998 MessagePreviewStore "retrieve" hit & miss metric

### PR DESCRIPTION
**Pull request by trantienduchn**
fast

### CODE REVIEW

**Comment by chibenwa** [comment]
@trantienduchn please rebase

**Comment by trantienduchn** [comment]
Rebase is really painful, I will wait till some are merged

File: grafana-reporting/MessageFastViewProjection-1575520507952.json [hunk 1]
```diff
@@ -0,0 +1,426 @@
+{
```
Comment (chibenwa) [inline grafana-reporting/MessageFastViewProjection-1575520507952.json]: Can you add documentation for this ? (at least in the README of that file...)

**Comment by trantienduchn** [comment]
```
[a7f9d5b8f770f91cc2f2fe58d271669d06d5738b] [ERROR]   MemoryJmapJamesServerTest$JamesServerJmapConfigurationTest$BadSecretKeyStore.jamesShouldNotStartWhenBadSecret:96 
[a7f9d5b8f770f91cc2f2fe58d271669d06d5738b] Expecting:
[a7f9d5b8f770f91cc2f2fe58d271669d06d5738b]   <com.google.inject.CreationException: Unable to create injector, see the following errors:
```

File: server/data/data-jmap-cassandra/pom.xml [hunk 2]
```diff
@@ -79,6 +79,11 @@
             <artifactId>testing-base</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (remk) [inline server/data/data-jmap-cassandra/pom.xml]: wrong order, put it before `testing-base`

**Comment by rouazana** [comment]
do you think you could add a ratio line in the third graph?

**Comment by trantienduchn** [comment]
> do you think you could add a ratio line in the third graph?

I tried, but I couldn't find a way

File: server/data/data-jmap-cassandra/pom.xml (hunk 2 above)
Comment (trantienduchn) [reply server/data/data-jmap-cassandra/pom.xml]: okay, it's simple, so I squashed

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**